fabtests/pytest/common.py

```diff
@@ -0,0 +1,241 @@
+import pytest
+
+
+def check_returncode(returncode, strict):
+    import errno
+
+    if returncode == 0:
+        return
+
+    if not strict:
+        if returncode == errno.ENODATA:
+            pytest.skip("ENODATA")
+            return
+
+        if returncode == errno.ENOSYS:
+            pytest.skip("ENOSYS")
+            return
+
+    error_msg = "returncode {}".format(returncode)
+    if returncode == 124:
+        error_msg += ", timeout"
+
+    pytest.fail(error_msg)
+
+class UnitTest:
+
+    def __init__(self, cmdline_args, base_command, is_negative=False):
+        self._cmdline_args = cmdline_args
+        self._base_command = base_command
+        self._is_negative = is_negative
+        self._command = cmdline_args.populate_command(base_command, "host")
+
+    def run(self):
+        import os
+        from tempfile import NamedTemporaryFile
+        from subprocess import Popen, TimeoutExpired
+
+        if self._cmdline_args.is_test_excluded(self._base_command, self._is_negative):
+            pytest.skip("excluded")
+            return
+
+        # start running
+        outfile = NamedTemporaryFile(prefix="fabtests_server.out.").name
+        process = Popen(self._command + ">& " + outfile, shell=True)
+
+        timeout = False
+        try:
+            process.wait(timeout=self._cmdline_args.timeout)
+        except TimeoutExpired:
+            process.terminate()
+            timeout = True
+
+        print("")
+        print("command: " + self._command)
+        print("stdout: ")
+        print(open(outfile).read())
+        os.unlink(outfile)
+
+        assert not timeout, "timed out"
+        check_returncode(process.returncode, self._cmdline_args.strict_fabtests_mode)
+
+class ClientServerTest:
+
+    def __init__(self, cmdline_args, executable, iteration_type=None, completion_type="transmit_complete", prefix_type="wout_prefix", datacheck_type="wout_datacheck", message_size=None):
```

Comment:
Long lines like this should be broken down. The full line is not even shown up in the review window.

Reply:
In new revision, I broke down this line and many other lines. I checked there are not lines longer than 120.